subprojects/testing-jvm/src/main/java/org/gradle/api/tasks/testing/Test.java

```diff
@@ -984,24 +959,32 @@ public TestFrameworkOptions options(Action<? super TestFrameworkOptions> testFra
         return options;
     }
 
+    /**
+     * Checks if the options for this test task have already been accessed; if so the framework cannot be changed on either the
+     * task or it's associated test suite.
+     *
+     * @since 8.0
+     */
+    @Internal
+    @Incubating
+    public Boolean getOptionsAccessed() {
```

Comment:
+1 for why we need an interface for test tasks + a private implementation.

Reply:
Yep.